app/src/hooks/api/useProjectApi.ts

```diff
@@ -105,16 +105,26 @@ const useProjectApi = (axios: AxiosInstance) => {
   /**
    * Upload project artifacts.
    *
-   * @param projectId
-   * @param files
-   * @return {*} {Promise<string[]>}
+   * @param {number} projectId
+   * @param {File[]} files
+   * @param {CancelTokenSource} [cancelTokenSource]
+   * @param {(progressEvent: ProgressEvent) => void} [onProgress]
+   * @return {*}  {Promise<string[]>}
    */
-  const uploadProjectArtifacts = async (projectId: number, files: File[]): Promise<string[]> => {
+  const uploadProjectArtifacts = async (
+    projectId: number,
+    files: File[],
+    cancelTokenSource?: CancelTokenSource,
+    onProgress?: (progressEvent: ProgressEvent) => void
+  ): Promise<string[]> => {
     const req_message = new FormData();
 
     files.forEach((file) => req_message.append('media', file));
 
-    const { data } = await axios.post(`/api/projects/${projectId}/artifacts/upload`, req_message);
+    const { data } = await axios.post(`/api/projects/${projectId}/artifacts/upload`, req_message, {
```

Comment:
question:
can we change this endpoint in the API to insert attachment metadata into the project_attachment table? so that it can be used in list view

Reply:
I think that work is part of its own ticket: https://quartech.atlassian.net/browse/BHBC-967